src/factories/node.ts

```diff
@@ -119,8 +117,18 @@ export async function nodeContainerFactory(node: RunGraphNode) {
   }
 
   function createArtifactsContainer(): void {
-    artifactsContainer = new Container()
-    container.addChild(artifactsContainer)
+    if (layout.isTemporal() && !settings.disableArtifacts) {
+      if (!artifactsContainer) {
+        artifactsContainer = new Container()
+      }
+
+      container.addChild(artifactsContainer)
```

Comment:
I did double check this. Adding something to a container that has already been added won't add a duplicate instance or anything. Pixi will shift it to the bottom of the children list, nbd though as we're generally assigning z-indexes for these anyway. Looked around to make sure toggling doesn't shift anything behind other things unintentionally and all looks good.